src/openvic-simulation/modifier/ModifierSum.cpp

```diff
@@ -19,38 +21,46 @@ bool ModifierSum::has_effect(ModifierEffect const& effect) const {
 	return value_sum.has_effect(effect);
 }
 
-void ModifierSum::add_modifier(Modifier const& modifier, fixed_point_t multiplier) {
-	modifiers[&modifier] += multiplier;
-	value_sum.multiply_add(modifier, multiplier);
+void ModifierSum::add_modifier(
+	Modifier const& modifier, modifier_source_t source, fixed_point_t multiplier, ModifierEffect::target_t targets
+) {
+	modifiers.push_back({ &modifier, multiplier, source, targets });
```

Comment:
Why use push_back here instead of emplace_back?
(I'm still learning when to use which)

Reply:
Usually `push_back` either copies (when given `value_type const&`) or moves (when given `value_type&&`) an already-constructed object into a container.
In contrast, `emplace_back` takes a list of arguments which are used to construct a new object, so `emplace_back(a, b, c)` is equivalent to `push_back(value_type { a, b, c })`. In some cases either function can be used, in particular via the copy or move constructor on the `emplace_back` side.

This specific case has changed to `modifiers.emplace_back(&modifier, multiplier, source, targets);` as we can use `emplace_back` to construct the object directly. Throughout the repo there are other objects which we can't normally construct using `emplace_back` as their constructors are private/protected and can only be called from a friend struct. This is because `emplace_back(a, b, c)` results in a constructor call from inside `emplace_back`, while `emplace_back({ a, b, c })` calls the constructor from the parent function before passing the object on to be move constructed.